File: PO_Projekt3/wilczejagody.py
from roslina import Roslina

class WilczeJagody(Roslina):
    def __init__(self, swiat, polozenie):
        super().__init__(swiat, polozenie, 99, 0, 0.10, 1)

    def kolizja(self, ofiara, atak):
        if atak:
            return

        if ofiara.getZyje():
            ofiara.zakonczZycie()

        if self._zyje:
            self.zakonczZycie()

    def tegoSamegoTypu(self, inny):
        return isinstance(inny, WilczeJagody)

    def stworzDziecko(self, cel):
        return WilczeJagody(self._swiat, cel)

    def rysowanie(self):
        return (133, 51, 255)

File: projekty/PO_Projekt2/ProgOb_Projekt2/src/main/java/pl/Dybowski/Adrian/Projekt2/WirtualnySwiat/Rosliny/WilczeJagody.java
package pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.Rosliny;

import pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.Organizm;
import pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.Stale;
import pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.Swiat;
import pl.Dybowski.Adrian.Projekt2.WirtualnySwiat.Wspolrzedne2D;

public class WilczeJagody extends Roslina {

	public WilczeJagody(Swiat swiat, Wspolrzedne2D polozenie) {
		super(swiat, polozenie, Stale.sila.get(WilczeJagody.class), Stale.inicjatywa.get(WilczeJagody.class),
				Stale.szansaRozprzestrzeniania.get(WilczeJagody.class), Stale.probyRozprzestrzeniania.get(WilczeJagody.class));
	}
	
	@Override
	public void kolizja(Organizm ofiara, boolean atak) {
		if (atak) return;
		
		if (ofiara.isZyje()) ofiara.zakonczZycie();
		
		if (zyje) zakonczZycie();
	}

	@Override
	public boolean tegoSamegoTypu(Organizm inny) {
		return inny instanceof WilczeJagody;
	}

	@Override
	public Organizm stworzDziecko(Wspolrzedne2D cel) {
		return new WilczeJagody(swiat, cel);
	}

}
